src/main/java/com/example/provider/utils/ServletUtils.java

```diff
@@ -49,15 +49,11 @@ public static void setErrorResponse(HttpServletResponse response, String message
   /**
    * Extracts the 'ID' piece of a URL request. Use for REST endpoints where ID is supplied in that
    * format.
-   *
-   * @throws IllegalArgumentException if path is not contained in the Request URL.
    */
-  public static String getEntityIdFromRequestPath(HttpServletRequest request)
-      throws IllegalArgumentException {
+  public static String getEntityIdFromRequestPath(HttpServletRequest request) {
```

Comment:
In case this method is used by other Servlets, could we please verify if this might have any side effects?

Reply:
After looking at the code within the other endpoints, we don't find getEntityIdFromRequestPath used at all besides in the VehicleServlet. The other endpoints like TripServlet call getPathInfo directly and do their own error checking. Additionally, we managed to successfully simulate a trip without anything breaking, so we should be good to go on that front.